packages/core/src/browser/statusbar/statusbar-entry-service.ts

```diff
@@ -0,0 +1,79 @@
+/*
+* Copyright (C) 2017 TypeFox and others.
+*
+* Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+*/
+import { injectable, inject } from "inversify";
+import { StatusbarEntry } from './statusbar';
+import { CommandService } from '../../common';
+
+export interface StatusbarEntryAttributes {
+    className?: string;
+    title?: string;
+    onclick?: () => void;
+    onmouseover?: () => void;
+    onmouseout?: () => void;
+}
+
+@injectable()
+export class StatusbarEntryService {
+
+    constructor( @inject(CommandService) protected readonly commands: CommandService) { }
+
+    createAttributes(entry: StatusbarEntry): StatusbarEntryAttributes {
```

Comment:
This logic can go back to StatusBarImpl as testing it is not worth the effort of making it independent of dom API.